engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -30,55 +33,132 @@ type backendSubscribeTransactions struct {
+// SendAndSubscribeTransactionStatuses sends a transaction and subscribes to its status updates.
+// It starts monitoring the status from the transaction's reference block ID.
+// If the transaction cannot be sent or an error occurs during subscription creation, a failed subscription is returned.
+func (b *backendSubscribeTransactions) SendAndSubscribeTransactionStatuses(
 	ctx context.Context,
 	tx *flow.TransactionBody,
 	requiredEventEncodingVersion entities.EventEncodingVersion,
 ) subscription.Subscription {
-	nextHeight, err := b.blockTracker.GetStartHeightFromBlockID(tx.ReferenceBlockID)
+	if err := b.sendTransaction(ctx, tx); err != nil {
+		b.log.Error().Err(err).Str("tx_id", tx.ID().String()).Msg("failed to send transaction")
+		return subscription.NewFailedSubscription(err, "failed to send transaction")
+	}
+
+	return b.createSubscription(ctx, tx.ID(), tx.ReferenceBlockID, 0, tx.ReferenceBlockID, requiredEventEncodingVersion, true)
+}
+
+// SubscribeTransactionStatusesFromStartHeight subscribes to the status updates of a transaction.
+// Monitoring starts from the specified block height.
+// If the block height cannot be determined or an error occurs during subscription creation, a failed subscription is returned.
+func (b *backendSubscribeTransactions) SubscribeTransactionStatusesFromStartHeight(
+	ctx context.Context,
+	txID flow.Identifier,
+	startHeight uint64,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+) subscription.Subscription {
+	return b.createSubscription(ctx, txID, flow.ZeroID, startHeight, flow.ZeroID, requiredEventEncodingVersion, false)
+}
+
+// SubscribeTransactionStatusesFromStartBlockID subscribes to the status updates of a transaction.
+// Monitoring starts from the specified block ID.
+// If the block ID cannot be determined or an error occurs during subscription creation, a failed subscription is returned.
+func (b *backendSubscribeTransactions) SubscribeTransactionStatusesFromStartBlockID(
+	ctx context.Context,
+	txID flow.Identifier,
+	startBlockID flow.Identifier,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+) subscription.Subscription {
+	return b.createSubscription(ctx, txID, startBlockID, 0, flow.ZeroID, requiredEventEncodingVersion, false)
+}
+
+// SubscribeTransactionStatusesFromLatest subscribes to the status updates of a transaction.
+// Monitoring starts from the latest block.
+// If the block cannot be retrieved or an error occurs during subscription creation, a failed subscription is returned.
+func (b *backendSubscribeTransactions) SubscribeTransactionStatusesFromLatest(
+	ctx context.Context,
+	txID flow.Identifier,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+) subscription.Subscription {
+	header, err := b.txLocalDataProvider.state.Sealed().Head()
 	if err != nil {
-		return subscription.NewFailedSubscription(err, "could not get start height")
+		b.log.Error().Err(err).Msg("failed to retrieve latest block")
```

Comment:
This should actually throw an exception since the node must have the current sealed block in storage.
e.g.
```golang
irrecoverable.Throw(ctx, err)
```